server/src/test/java/org/apache/druid/segment/metadata/SegmentSchemaCacheTest.java

```diff
@@ -80,11 +80,23 @@ public void testCacheTemporaryMetadataQueryResults()
     Assert.assertTrue(schema.isPresent());
     Assert.assertEquals(expected, schema.get());
 
-    cache.resetTemporaryPublishedMetadataQueryResultOnDBPoll();
+    // simulate call after segment polling
 
-    Assert.assertFalse(cache.isSchemaCached(id));
+    ImmutableMap.Builder<SegmentId, SegmentMetadata> segmentMetadataBuilder = ImmutableMap.builder();
+    segmentMetadataBuilder.put(id, new SegmentMetadata(5L, "fp"));
+
+    ImmutableMap.Builder<String, SchemaPayload> schemaPayloadBuilder = ImmutableMap.builder();
+    schemaPayloadBuilder.put("fp", new SchemaPayload(rowSignature));
+
+    SegmentSchemaCache.FinalizedSegmentSchemaInfo finalizedSegmentSchemaInfo =
+        new SegmentSchemaCache.FinalizedSegmentSchemaInfo(segmentMetadataBuilder.build(), schemaPayloadBuilder.build());
+
+    cache.updateFinalizedSegmentSchema(finalizedSegmentSchemaInfo);
+
+    Assert.assertNull(cache.getTemporaryPublishedMetadataQueryResults(id));
```

Comment:
We should have one ut where cache.getTemporaryPublishedMetadataQueryResults(id2) where id2 is not present in db poll but got back filled.